include/metrics/Timer.h

```diff
@@ -23,28 +22,20 @@ template<
 class Timer
 {
 public:
+    using this_type = Timer< METRIC_TYPE, TIME_UNIT >;
     using metric_type = METRIC_TYPE;
-    using duration_type = typename std::chrono::duration< double, typename TIME_UNIT::period >;
+    using timer_type = ScopeTimer< TIME_UNIT, double >;
+    using duration_type = typename timer_type::duration_type;
 
 private:
-    metric_type* _timer;
-    timespec _start;
+    std::unique_ptr< timer_type > _timer;
 
 public:
     explicit Timer(metric_type* timer = nullptr,
```

Comment:
timer being prometheus::Histogram and _time being ScopeTimer is a bit confusing for review, at least.

Reply:
perfect!